hyperapp.d.ts

```diff
@@ -54,94 +56,68 @@ export type ActionResult<State extends Hyperapp.State> =
   | null
   | void
 
-export type ActionCallers<
-  State extends Hyperapp.State,
-  Actions extends Hyperapp.Actions<State>
-> = {
-  [P in keyof Actions]: (
-    data: any
-  ) => ActionResult<State> | ActionCallers<State, Actions>
-}
 
-export interface Actions<State extends Hyperapp.State> {
+export interface Actions {
   [action: string]:
-    | Actions<State>
+    | Actions
     | ((
-        state: State,
-        actions: ActionCallers<State, Actions<State>>,
         data: any
       ) => ActionResult<State>)
 }
 
-export interface Events<
+export type ActionsImpl<
```

Comment:
I'm still playing around with it (and understanding #385) but I think if we change `ActionsImpl` to the following, it will correctly type fragmented actions and we *might* be able to do something similar for fragmented states:
```ts
export type ActionsImpl<
State extends Hyperapp.State,
Actions
> = {
[P in keyof Actions]: ((
  state: State,
  actions: Actions,
  data: any
) => ActionResult<State>) | ActionsImpl<State, Actions[P]>
}
```

Reply:
@andrewiggins I did in on the other PR by changing the interface of State, once we'll have it all fixed/merged over there I'll update this PR.